### PR TITLE
Add "unit" tests checking that shipped rulesets load

### PR DESCRIPTION
**Pull request by lmoureaux**
Also made me fix issues with the server CLI.

The tests pass on Windows when run from the MSYS terminal, but not in Github actions. The issue is that the server never reacts to the `quit` command sent by the test; `stdin` on Windows is a disaster.

### CODE REVIEW

**Comment by lmoureaux** [comment]
The tests fail on Windows because of #850 (most likely)

**Comment by lmoureaux** [comment]
> The tests fail on Windows because of https://github.com/longturn/freeciv21/issues/850 (most likely)

Looks like a different issue.

File: data/tests/rulesets.cpp.in [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Tries to spawn a server with a ruleset.
+ */
+void test_rulesets::load()
+{
+  QFETCH(QString, name);
+  QFETCH(bool, exists);
+
+  QProcess p;
+  p.start(SERVER_PATH, {QStringLiteral("-r"), name});
```
Comment (zekoz) [inline data/tests/rulesets.cpp.in:46]: Does this fall back to the default ruleset on failure to load?

**Review by zekoz — APPROVED** [review]

File: data/tests/rulesets.cpp.in (hunk 1 above)
Comment (lmoureaux) [reply data/tests/rulesets.cpp.in:46]: Not anymore, the server now exits if any failing command in the startup script. This is checked in the `cli` test.

**Pull request merged**